### PR TITLE
new device, temperature gate

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
This PR adds a new RPD device, the temperature gate. This is similar to the pressure valve, where you set a temperature in the gui and if the gas before the gate has a temperature lower than what you set, it flows, otherwise it stays put. You can also use a multitool on the gate to invert it and so the gas will flow if the temperature is Higher than the set one.

## Why It's Good For The Game
More atmos autism devices, help streamline productions and control of gases (very useful when making H2)

## Changelog
:cl:
add: New RPD device, the temperature gate
/:cl:

### CODE REVIEW

**Comment by nightred** [comment]
well this makes h2 much simpler to make, no more oppies fires because the gasses got hot.

**Comment by Donkie** [comment]
Just make ingame programmable atmos devices for ultimate atmosia revolution

**Comment by Ghilker** [comment]
.... oh boy

**Comment by LemonInTheDark** [comment]
> 
> 
> Just make ingame programmable atmos devices for ultimate atmosia revolution

Nice try circuit fanatic

File: code/modules/atmospherics/machinery/components/binary_devices/temperature_gate.dm [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/obj/machinery/atmospherics/components/binary/temperature_gate
+	icon_state = "tgate_map-3"
+	name = "temperature gate"
+	desc = "An activable one way valve that let gas pass through if the pressure on the input side is higher than the set pressure."
```
Comment (Iatots) [inline code/modules/atmospherics/machinery/components/binary_devices/temperature_gate.dm:4]: Oi

Comment (Ghilker) [reply code/modules/atmospherics/machinery/components/binary_devices/temperature_gate.dm:4]: oof thanks for the tip

File: tgui/packages/tgui/interfaces/AtmosTempGate.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        <Section>
+          <LabeledList>
+            <LabeledList.Item label="Power">
+              <Button
+                icon={data.on ? 'power-off' : 'times'}
+                content={data.on ? 'On' : 'Off'}
+                selected={data.on}
+                onClick={() => act('power')} />
+            </LabeledList.Item>
+            {(
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/AtmosTempGate.js:21]: What is this, delete.

**Pull request merged**